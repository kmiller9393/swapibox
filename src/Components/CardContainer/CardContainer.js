import React from 'react';
import { Card } from '../Card/Card.js';
import PropTypes from 'prop-types';

export const CardContainer = ({ selectedGroup, favoriteItem }) => {
  const groupData = selectedGroup.map((item, index) => (
    <Card {...item} favoriteItem={favoriteItem} key={index} />
  ));

  return <div className="card-container">{groupData}</div>;
};

const { array, func } = PropTypes;
CardContainer.propTypes = {
  selectedGroup: array,
  favoriteItem: func
};
