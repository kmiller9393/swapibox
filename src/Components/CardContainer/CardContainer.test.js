import React from 'react';
import { shallow } from 'enzyme';

import { CardContainer } from './CardContainer';

describe('CardContainer', () => {
  let mockSelectedGroup;
  let mockFavoriteItem;
  let wrapper;
  let favoriteItem;

  beforeEach(() => {
    mockSelectedGroup = [{
      Class: "wheeled",
      Model: "Digger Crawler",
      Passengers: "30",
      name: "Sand Crawler"}];
    mockFavoriteItem = jest.fn();
    wrapper = shallow(<CardContainer selectedGroup={mockSelectedGroup} favoriteItem={mockFavoriteItem}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});