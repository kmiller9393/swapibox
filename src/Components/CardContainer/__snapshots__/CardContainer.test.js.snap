// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favoriteItem={[Function]}
    selectedGroup={
        Array [
            Object {
              "Class": "wheeled",
              "Model": "Digger Crawler",
              "Passengers": "30",
              "name": "Sand Crawler",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          Class="wheeled"
          Model="Digger Crawler"
          Passengers="30"
          favoriteItem={[Function]}
          name="Sand Crawler"
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "Class": "wheeled",
          "Model": "Digger Crawler",
          "Passengers": "30",
          "favoriteItem": [Function],
          "name": "Sand Crawler",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            Class="wheeled"
            Model="Digger Crawler"
            Passengers="30"
            favoriteItem={[Function]}
            name="Sand Crawler"
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "Class": "wheeled",
            "Model": "Digger Crawler",
            "Passengers": "30",
            "favoriteItem": [Function],
            "name": "Sand Crawler",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
